src/App.tsx

```diff
@@ -1,26 +1,353 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
+import {
+  USER_ID,
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateCompletedTodo,
+  updateTitleTodo,
+} from './api/todos';
+import { ToDoHeader } from './components/todoapp_header';
+import { TodoList } from './components/todoapp_todolist';
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+import { ErrorType } from './types/ErrorType';
+import { Errors } from './components/todoapp_error';
 
-const USER_ID = 0;
+type TempTodo = {
+  id: number;
+  completed: boolean;
+  title: string;
+  userId: number;
+};
 
 export const App: React.FC = () => {
+  const [tasks, setTasks] = useState<Todo[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [status, setStatus] = useState<Status>(Status.all);
+  const [taskTitle, setTaskTitle] = useState('');
+  const [allDone, setAllDone] = useState(false);
+  const [IsSubmitting, setIsSubmitting] = useState(false);
+  const [tempTodo, setTempTodo] = useState<TempTodo | null>(null);
+  const [deletingIds, setDeletingIds] = useState<number[]>([]);
+  const [isUpdating, setIsUpdating] = useState<number[] | []>([]);
+  const [isEditing, setIsEditing] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState('');
+  const completedTodos = tasks.filter(todo => todo.completed);
+  // const activeTodos = tasks.filter(todo => !todo.completed);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const errorType: ErrorType = {
+    empty: 'Title should not be empty',
+    load: 'Unable to load todos',
+    add: 'Unable to add a todo',
+    found: 'Todo not found',
+  };
+
+  const taskLeft = tasks?.filter(task => !task.completed).length;
+
+  const focusInput = () => {
+    setTimeout(() => {
+      inputRef.current?.focus();
+    }, 0);
+  };
+
+  const filterTodo: (todos: Todo[], mode: Status) => Todo[] = (todos, mode) => {
+    switch (mode) {
+      case Status.completed:
+        return todos.filter(task => task.completed);
+      case Status.active:
+        return todos.filter(task => !task.completed);
+      case Status.all:
+      default:
+        return tasks;
+    }
+  };
+
+  const filteredTodos = filterTodo(tasks, status);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleError: (errorMsg: string) => void = errorMsg => {
+    setErrorMessage(errorMsg);
+  };
+
+  const addNewTodo = () => {
+    if (!taskTitle.trim()) {
+      handleError(errorType.empty);
+
+      return Promise.reject();
+    }
+
+    setTempTodo({ id: 0, completed: false, title: taskTitle, userId: USER_ID });
```

Comment:
split it

Reply:
![wyyys](https://github.com/mate-academy/react_todo-app-with-api/assets/138024608/c012b445-d6be-4d1a-9cd9-a86b17b78f97)
![wyyysblad](https://github.com/mate-academy/react_todo-app-with-api/assets/138024608/7de81ac8-94d9-4527-a305-221b55c70250)
 code have pass all tests at pc but on git is showing fail. why is that happen?